lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -8,6 +10,58 @@
 from ..population.dwelltime import DwelltimeModel
 
 
+def _read_txt(testfile, delimiter):
+    """Read text-based csv file with KymoTrack data
+
+    Parameters
+    ----------
+    delimiter : str
+        Delimiter to use.
+
+    Raises
+    ------
+    IOError
+        If the file is in the wrong format.
+    """
+    if isinstance(testfile, os.PathLike):
+        testfile = os.fspath(testfile)
+
+    if isinstance(testfile, str):
+        with open(testfile, "r") as f:
+            header_lines = [f.readline(), f.readline()]
+    else:
+        # Direct StringIO
+        header_lines = [testfile.readline(), testfile.readline()]
```

Comment:
That would be breaking. Also, we have people depending on that API.

Reply:
The old one supported reading from `StringIO` since we directly forwarded that to `numpy`. Although you are correct that it is private API. I have reworked it a little with `pathlib`. Hopefully that makes it a little less ugly.

Reply:
yeah I think it's a lot clearer using `pathlib`

Reply:
Yeah, that was a nice suggestion!